File: apple/Sources/ReactNativeWatchOS/Modifiers/Kerning.swift
import SwiftUI

/// SwiftUI `.kerning(_:)`. Sets the spacing between each character pair,
/// in points. Registered in the generic registry so it applies to Text
/// and any view with inheritable typography. Defaults to `0` when the
/// value is absent (SwiftUI's no-kerning default).
enum RNWKerningModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("kerning") { view, params, _ in
            AnyView(view.kerning(params.cgFloat("value") ?? 0))
        }
    }
}

File: src/modifiers/kerning.ts
import { createModifier } from './createModifier';

/// SwiftUI `.kerning(_:)`. Sets the spacing, in points, between each
/// character pair. Positive values add space; negative values tighten.
/// Unlike `tracking`, `kerning` does not affect the trailing edge of the
/// last character.
export function kerning(value: number) {
  return createModifier('kerning', { value });
}
